Panic on resource configurator when Cloud SQL database fails to create

<!--
Thank you for helping to improve Official GCP Provider!

Please be sure to search for open issues before raising a new one. We use issues
for bug reports and feature requests.
-->

### What happened?
A panic occurs on the sql resource configurator when a new Cloud SQL database fails to create. The specific error for starting the database was that there was no IP space available on the network.
This should fail gracefully and produce an event / status condition on the resource.

It looks like the error is [here](https://github.com/upbound/provider-gcp/blob/main/config/sql/config.go#L59), trying to get a certificate that doesn't exist, since the instance failed to create.

```
panic: runtime error: index out of range [0] with length 0 [recovered]
	panic: runtime error: index out of range [0] with length 0

goroutine 1618 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
	sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:119 +0x1fa
panic({0x27089a0, 0xc001e317b8})
	runtime/panic.go:884 +0x212
github.com/upbound/provider-gcp/config/sql.Configure.func1.1(0xc00205c030?)
	github.com/upbound/provider-gcp/config/sql/config.go:59 +0x645
github.com/upbound/upjet/pkg/resource.GetConnectionDetails(0x7f225e7382c0?, {0x7f225e7383e8?, 0xc000d56380?}, 0xc002a76500)
	github.com/upbound/upjet@v0.9.0-rc.0.0.20230427131841-1d0b7b0c08f8/pkg/resource/sensitive.go:85 +0x87
github.com/upbound/upjet/pkg/controller.(*external).Observe(0xc00204c780, {0x2f93810, 0xc001f50de0}, {0x2fcdc80, 0xc000d56380})
	github.com/upbound/upjet@v0.9.0-rc.0.0.20230427131841-1d0b7b0c08f8/pkg/controller/external.go:206 +0x599
github.com/crossplane/crossplane-runtime/pkg/reconciler/managed.(*Reconciler).Reconcile(0xc001a29800, {0x2f93848, 0xc002024210}, {{{0x0?, 0x7f225e8614b8?}, {0xc001e0eb60?, 0x7f22860315b8?}}})
	github.com/crossplane/crossplane-runtime@v0.20.0-rc.0.0.20230406155702-4e1673b7141f/pkg/reconciler/managed/reconciler.go:805 +0x293f
github.com/crossplane/crossplane-runtime/pkg/ratelimiter.(*Reconciler).Reconcile(0xc002b9e3c0, {0x2f93848, 0xc002024210}, {{{0x0?, 0x10?}, {0xc001e0eb60?, 0x40da67?}}})
	github.com/crossplane/crossplane-runtime@v0.20.0-rc.0.0.20230406155702-4e1673b7141f/pkg/ratelimiter/reconciler.go:54 +0x16b
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x2f93848?, {0x2f93848?, 0xc002024210?}, {{{0x0?, 0x23a65e0?}, {0xc001e0eb60?, 0x0?}}})
	sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:122 +0xc8
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc000ba0500, {0x2f937a0, 0xc0003d7c00}, {0x25997e0?, 0xc001bb78c0?})
	sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:323 +0x38f
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc000ba0500, {0x2f937a0, 0xc0003d7c00})
	sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:274 +0x1d9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
	sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:235 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
	sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:231 +0x333
```


### How can we reproduce it?
Create a database instance with no IP address space available in the network like such:
```yaml
apiVersion: sql.gcp.upbound.io/v1beta1
kind: DatabaseInstance
metadata:
  name: my-instance
spec:
  providerConfigRef:
    name: sql-config
  forProvider:
    project: my-gcp-project
    databaseVersion: POSTGRES_15
    region: asia-northeast1
    rootPasswordSecretRef:
      key: password
      name: "root-postgres"
      namespace: my-namespace
    settings:
      - tier: db-g1-small
        ipConfiguration:
          - ipv4Enabled: true
            privateNetwork: projects/my-gcp-project/global/networks/my-network
            requireSsl: true
        availabilityType: REGIONAL
```

Wait until the instance fails to create, see panic in the sql provider.

### What environment did it happen in?

* Crossplane Version: :v1.12.2
* Provider Version: v0.34.0 (using provider family to get sql provider specifically)
* Kubernetes Version: 
```
Client Version: version.Info{Major:"1", Minor:"26", GitVersion:"v1.26.1", GitCommit:"8f94681cd294aa8cfd3407b8191f6c70214973a4", GitTreeState:"clean", BuildDate:"2023-01-18T15:58:16Z", GoVersion:"go1.19.5", Compiler:"gc", Platform:"darwin/arm64"}
Server Version: version.Info{Major:"1", Minor:"26", GitVersion:"v1.26.5-gke.1200", GitCommit:"4eb57372b0f0ac0023caca078161b75185febeef", GitTreeState:"clean", BuildDate:"2023-06-01T19:55:11Z", GoVersion:"go1.19.9 X:boringcrypto", Compiler:"gc", Platform:"linux/amd64"}
```
* Kubernetes Distribution: GKE

--- a/config/sql/config.go
+++ b/config/sql/config.go
@@ -55,7 +55,7 @@ func Configure(p *config.Provider) { //nolint:gocyclo
 				conn[xpv1.ResourceCredentialsSecretPasswordKey] = []byte(a)
 			}
 			// map
-			if certSlice, ok := attr["server_ca_cert"].([]interface{}); ok {
+			if certSlice, ok := attr["server_ca_cert"].([]interface{}); ok && len(certSlice) > 0 {
 				if certattrs, ok := certSlice[0].(map[string]interface{}); ok {
 					if a, ok := certattrs["cert"].(string); ok {
 						conn[CloudSQLSecretServerCACertificateCertKey] = []byte(a)

